File: JavaScript/프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    let answer = 1
    
    for(let v of arr) {
        answer = lcm(answer, v)
    }
    
    return answer
}

function gcd(a, b) {
    if ( b == 0 ) return a
    return gcd(b, a%b)
}

function lcm(a, b) {
    return a*b / gcd(a, b)
}

File: Kotlin/프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.kt
class Solution {
    fun solution(arr: IntArray): Int {
        var answer = arr[0]
        arr.forEach {answer = lcm(answer, it)}
        return answer
    }
    
    fun gcd(a:Int, b:Int): Int  = if(b != 0) gcd(b, a % b) else a
    fun lcm(a:Int, b:Int): Int = a*b / gcd(a,b)
}